### PR TITLE
fix the problem when the camera of the scene is not OrthographicCamer…

### PR DESCRIPTION
**Pull request by appache163**
fix the problem when the camera of the scene is not OrthographicCamer and PerspectiveCamera, it will make a null mistake

### CODE REVIEW

**Comment by appache163** [comment]
fix the problem when the camera of the scene is not OrthographicCamera and PerspectiveCamera, it will make a null mistake

File: examples/js/loaders/GLTFLoader.js [hunk 1]
```diff
@@ -3385,6 +3385,11 @@
 
 			}
 
+			if ( ! camera ) {
```
Comment (Mugen87) [inline examples/js/loaders/GLTFLoader.js:3388]: According to the glTF spec, the projection can only be perspective or orthographic. How can `cameraDef.type` hold a different value without being invalid glTF?

**Comment by ycw** [comment]
more ideal to have exhaustive check for `cameraDef.type`, similar to `primitive.mode`
https://github.com/mrdoob/three.js/blob/9c8ede023ccc2ca7810b46025703a263a4a41e06/examples/jsm/loaders/GLTFLoader.js#L3488-L3496

**Comment by donmccurdy** [comment]
@ycw In general, we don't intend to do input validation in three.js. For one thing, error messages consist mostly of large strings, and those strings can't be minified. In other cases it may cost us on performance. The `primitive.mode` example could be removed, or not, but we're certainly not setting a goal of adding more validation errors of this type.

Moreover – throwing errors like this prevents our users from adding support for new types in the extension API, where returning null might allow that.

I'll close the PR for now; but glad to reopen if the use case is something other than an incorrectly-formed input file.

**Comment by ycw** [comment]
@donmccurdy its not about input validation, but **project diagnostics**, "null mistake" is contagious :|

**Comment by ycw** [comment]
@donmccurdy its worth to diagnose [4k loc](https://github.com/mrdoob/three.js/blob/dev/examples/jsm/loaders/GLTFLoader.js) by tools, and tools need hints like exhaustiveness checking. We understand `cameraDef.type` is either 'perspective' or 'orthographic', but tools don't; for this case, a if-else is sufficient to reflect its binary nature:

```js
let camera
if ( cameraDef.type === 'perspective' ) { 
  camera = new PerspectiveCamera(..) 
} else { 
  camera = new OrthographicCamera(..) 
}
// here, camera must not be undefined
```

this would elim. OP's "nullish mistake" at L3575, and in turn avoids fn from contagion at L3577 

https://github.com/mrdoob/three.js/blob/9c8ede023ccc2ca7810b46025703a263a4a41e06/examples/jsm/loaders/GLTFLoader.js#L3575-L3577

so,

> What project needs this diagnostic information, and why does that information need to come from THREE.GLTFLoader? GLTFLoader fails on invalid input, as intended. Having a handwritten error message describing every failure is a non-goal here, sorry.

its not about input validation, nor error handling strategy, its about satisfying static analyzer to improve GLTFLoader itself. :)

**Comment by donmccurdy** [comment]
Sorry, I think we'll have to disagree on this. As a maintainer of this file, the changes proposed create other problems. 

I'm willing to make a change here _if_ users need to support a custom camera extension that requires the change. Otherwise let's leave this as-is.

**Comment by appache163** [comment]
Thank you all for your reply, I'm making a rough draft for a custom GLTF, I found this code may cause an error，so I raise this question，I'll make a test for this，thanks!

**Comment by ycw** [comment]
@donmccurdy would be great if clarifies by a comment, similar to L1610 :)

https://github.com/mrdoob/three.js/blob/9c8ede023ccc2ca7810b46025703a263a4a41e06/examples/jsm/loaders/GLTFLoader.js#L1598-L1611

**Pull request closed without merging**